packages/site-admin/src/components/sidebar-button/index.tsx

```diff
@@ -0,0 +1,23 @@
+/**
+ * External dependencies
+ */
+import { Button } from '@wordpress/components';
+import clsx from 'clsx';
+/**
+ * Internal dependencies
+ */
+import './style.scss';
+/**
+ * Types
+ */
+import type { ButtonProps } from '@wordpress/components/build-types/button/types';
```

Comment:
Instead of importing from this path (which I'm not even sure it's a canonical way), we could do something like

`React.ComponentProps< typeof Button >`